libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationRerouteInterruptionTest.kt

```diff
@@ -0,0 +1,91 @@
+package com.mapbox.navigation.core
+
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.reroute.NavigationRerouteController
+import com.mapbox.navigation.core.reroute.RerouteController
+import com.mapbox.navigation.core.reroute.RerouteState
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.InternalCoroutinesApi
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.ParameterizedRobolectricTestRunner
+import org.robolectric.annotation.Config
+
+@ExperimentalPreviewMapboxNavigationAPI
+@Config(shadows = [ShadowReachabilityFactory::class])
+@InternalCoroutinesApi
+@ExperimentalCoroutinesApi
+@RunWith(ParameterizedRobolectricTestRunner::class)
+internal class MapboxNavigationRerouteInterruptionTest(
+    private val previousState: RerouteState?,
+    private val expectedNumberOfInterruptions: Int,
+) : MapboxNavigationBaseTest() {
+
+    companion object {
+
+        @JvmStatic
+        @ParameterizedRobolectricTestRunner.Parameters
+        fun data(): List<Array<Any?>> {
+            return listOf<Array<Any?>>(
+                arrayOf(null, 0),
+                arrayOf(RerouteState.Idle, 0),
+                arrayOf(RerouteState.Interrupted, 0),
+                arrayOf(RerouteState.FetchingRoute, 1),
+                arrayOf(mockk<RerouteState.Failed>(), 0),
+                arrayOf(mockk<RerouteState.RouteFetched>(), 0),
+            ).apply {
+                assert(size == RerouteState::class.sealedSubclasses.size + 1) // 1 is for null
```

Comment:
it looks redundant, as the number of cases might be non-equal to amount of sealed classes items
```suggestion
```

Reply:
Why? The state is checked. The state has `RerouteState` type. And it's sealed. It's just something that I like to do in order not to forget about the tests when a new subclass/enum value, etc. is added. You never know which tests check every possible value. It's nice to think whether this new value should affect this or that situation.

Reply:
My point was that sometimes we need more than 1 case for 1 state. 

Also, the sealed class won't extend for current major version as it will break SEMVER

Reply:
> My point was that sometimes we need more than 1 case for 1 state.

But then we will probably have _n_ cases for each state and then it will be `n * size + 1`. :)

Reply:
amount of cases can be different for each state 😉 

anyway 👇 
> Also, the sealed class won't extend for current major version as it will break SEMVER